src/buildMiddleware.ts

```diff
@@ -1,98 +1,167 @@
-        if (!provided) {
-          continue;
-        }
+    }
 
-        let invalidateSubscriptions =
-          (entity.id !== undefined
-            ? // id given: invalidate all queries that provide this type & id
-              provided[entity.id]
-            : // no id: invalidate all queries that provide this type
-              Object.values(provided).flat(1)) ?? [];
+    if (unsubscribeQueryResult.match(action)) {
+      handleUnsubscribe(action.payload, api);
+    }
 
-        for (const invalidate of invalidateSubscriptions) {
-          (toInvalidate[invalidate.endpoint] ??= new Set()).add(invalidate.serializedQueryArgs);
-        }
+    if (updateSubscriptionOptions.match(action)) {
+      handlePolling(action.payload, api);
+    }
+    if (queryThunk.pending.match(action) || (queryThunk.rejected.match(action) && action.meta.condition)) {
+      handlePolling(action.meta.arg, api);
+    }
+
+    return result;
+  };
+
+  return { middleware };
+
+  function invalidateEntities(entities: readonly FullEntityDescription<string>[], api: Api) {
+    const rootState = api.getState();
+    const state = rootState[reducerPath] as InternalState;
+
+    const toInvalidate: { [endpoint: string]: Set<string> } = {};
+    for (const entity of entities) {
+      const provided = state.provided[entity.type];
+      if (!provided) {
+        continue;
       }
 
-      batch(() => {
-        for (const [endpoint, collectedArgs] of Object.entries(toInvalidate)) {
-          for (const serializedQueryArgs of collectedArgs) {
-            const querySubState = (state.queries as QueryState<any>)[endpoint]?.[serializedQueryArgs];
-            if (querySubState) {
-              if (querySubState.subscribers.length === 0) {
-                api.dispatch(removeQueryResult({ endpoint, serializedQueryArgs }));
-              } else if (querySubState.status !== QueryStatus.uninitialized) {
-                const startQuery = queryActions[endpoint];
-                const arg = querySubState.arg;
-                api.dispatch(startQuery(arg, { subscribe: false, forceRefetch: true }));
-              } else {
-              }
+      let invalidateSubscriptions =
+        (entity.id !== undefined
+          ? // id given: invalidate all queries that provide this type & id
+            provided[entity.id]
+          : // no id: invalidate all queries that provide this type
+            Object.values(provided).flat(1)) ?? [];
+
+      for (const invalidate of invalidateSubscriptions) {
+        (toInvalidate[invalidate.endpoint] ??= new Set()).add(invalidate.serializedQueryArgs);
+      }
+    }
+
+    batch(() => {
+      for (const [endpoint, collectedArgs] of Object.entries(toInvalidate)) {
+        for (const serializedQueryArgs of collectedArgs) {
+          const querySubState = (state.queries as QueryState<any>)[endpoint]?.[serializedQueryArgs];
+          // const querySubState = querySelectors[endpoint]?.(serializedQueryArgs)(rootState);
```

Comment:
This breaks, because the selector calls serializeQueryParams again - causing things like "\"1"\" instead of "1".